internal/repos/syncer.go

```diff
@@ -262,20 +265,37 @@ func (s *Syncer) SyncRepo(ctx context.Context, name api.RepoName) (repo *types.R
 	}
 
 	if repo == nil {
-		// We don't have this repo yet, so block before returning.
-		return s.syncRepo(ctx, codehost, name, nil)
+		// We don't have this repo yet so we need to try and sync it now and we will block
+		// until syncing is complete
```

Comment:
```suggestion
		// until syncing is complete.
```

:stuck_out_tongue_closed_eyes: